loquacious-credenza/Lantern
# Lockscreen

#### What's this PR do?

This adds a passcode screen so that checkin can be authenticated.  The authentication screen loads while the password has not been set.  Once the password is set it is written in AsyncStorage and persisted on the server.

When the application is reloaded, the password is read from AsyncStorage and read from the server as fallback.

When a trip is started it will be in tracking mode until proximity detection triggers the appearance of the checkin button.  When the button is clicked the user must authenticate to cancel the trip.
#### What are the important parts of the code?

Take a look at components `passwordSet`, `passwordConfirm`.  The `user` reducer has the password updating and there changes to logging action and `setPassword` action.

Console.logs are still present for debugging purposes.
#### How should this be tested by the reviewer?

Clear AsyncStorage, clear the user from the db.  Load the app, login, and see the authentication screen show up when the map view loads.  Enter a new password, create a trip and start tracking. Reload the app and see that the map is loaded in tracking state (without going to authentication page).  Checkin on the trip and then reload the app.  There is no active trip but the password is already set.
